tests/test_nhg.py

```diff
@@ -216,6 +287,358 @@ def test_route_nhg(self, dvs, dvs_route, testlog):
+        asic_nhs_count = len(asic_db.get_keys(self.ASIC_NHS_STR))
+
+        # Add a group containing labeled weighted NHs
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('mpls_nh', 'push1,push3'),
+                                            ('ifname', 'Ethernet0,Ethernet4'),
+                                            ('weight', '2,4')])
+        nhg_ps.set('group1', fvs)
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + 1)
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+
+        # NhgOrch should create two next hops for the labeled ones
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 2)
+
+        # Assert the weights are properly set
+        nhgm_ids = self.get_nhgm_ids('group1', dvs)
+        weights = []
+        for k in nhgm_ids:
+            fvs = asic_db.get_entry(self.ASIC_NHGM_STR, k)
+            weights.append(fvs['SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT'])
+        assert set(weights) == set(['2', '4'])
+
+        # Create a new single next hop with the same label
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1'),
+                                          ('mpls_nh', 'push1'),
+                                            ('ifname', 'Ethernet0')])
+        nhg_ps.set('group2', fvs)
+
+        # No new next hop should be added
+        time.sleep(1)
+        assert len(asic_db.get_keys(self.ASIC_NHS_STR)) == asic_nhs_count + 2
+
+        # Create a new single next hop with a different label
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1'),
+                                          ('mpls_nh', 'push2'),
+                                            ('ifname', 'Ethernet0')])
+        nhg_ps.set('group3', fvs)
+
+        # A new next hop should be added
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 3)
+
+        # Delete group3
+        nhg_ps._del('group3')
+
+        # Group3's NH should be deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 2)
+
+        # Delete group2
+        nhg_ps._del('group2')
+
+        # The number of NHs should be the same as they are still referenced by
+        # group1
+        time.sleep(1)
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 2)
+
+        # Update group1 with no weights and both labeled and unlabeled NHs
+        fvs = swsscommon.FieldValuePairs([('nexthop', '10.0.0.1,10.0.0.3'),
+                                          ('mpls_nh', 'push2,na'),
+                                            ('ifname', 'Ethernet0,Ethernet4')])
+        nhg_ps.set('group1', fvs)
+
+        # Group members should be replaced and one NH should get deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, asic_nhgms_count + 2)
+        asic_db.wait_for_n_keys(self.ASIC_NHS_STR, asic_nhs_count + 1)
+
```

Comment:
Also check the SAI_NHOP_ID does not change.

Reply:
Do you mean the group's SAI ID or the group member's NEXT_HOP_GROUP_ID?